dev/insights/proxy/flaskapp.py

```diff
@@ -0,0 +1,326 @@
+#!/usr/bin/env python
+
+###############################################################################
+#
+#   local insights mitm proxy
+#
+#       Proxies requests from a client to the api server in a way that
+#       simulates how the insights platform handles requests. Clients will
+#       authenticate to the proxy with a username+pass or a token and then
+#       the proxy will send the request to the api along with the appropriate
+#       X_RH_IDENTITY header that is required for authentication.
+#
+###############################################################################
+
+
+import base64
+import json
+import os
+import uuid
+import requests
+import tempfile
+from pprint import pprint
+from requests_toolbelt.multipart.encoder import MultipartEncoder
+
+import flask
+from flask import Flask
+from flask import jsonify
+from flask import request
+
+
+pprint(os.environ)
+
+
+FILE_CACHE_DIR = '/tmp/flask-filecache'
+SERVER_BASE_URL = os.environ.get('SERVER_BASE_URL', 'http://localhost:8080')
+UPSTREAM_PROTO = os.environ.get('UPSTREAM_PROTO', 'http')
+UPSTREAM_HOST = os.environ.get('UPSTREAM_HOST', 'localhost')
+UPSTREAM_PORT = os.environ.get('UPSTREAM_PORT', '5001')
+UPSTREAM_BASE_URL = f'{UPSTREAM_PROTO}://{UPSTREAM_HOST}:{UPSTREAM_PORT}'
+print(f'UPSTREAM_BASE_URL: {UPSTREAM_BASE_URL}')
+
+
+# Refresh tokens are what the clients use to request a bearer token from keycloak
+REFRESH_TOKENS = {
+    '1234567890': '1',
+    '1234567891': '2',
+    '1234567892': '3',
+    '1234567893': '4',
+}
+
+# Bearer tokens are used for all api calls
+BEARER_TOKENS = {}
+
+# This set of users mimics what is found in the ephemeral environments.
+USERS = {
+    '1': {
+        'account_number': 6089719,
+        'user': {
+            'username': 'jdoe',
+            'email': 'jdoe@redhat.com',
+            'first_name': 'john',
+            'last_name': 'doe'
+        }
+    },
+    '2': {
+        'account_number': 6089723,
+        'user': {
+            'username': 'iqe_normal_user',
+            'email': 'iqe_normal_user@redhat.com',
+            'first_name': '',
+            'last_name': ''
+        }
+    },
+    '3': {
+        'account_number': 6089720,
+        'user': {
+            'is_org_admin': True,
+            'username': 'org-admin',
+            'email': 'org-admin@redhat.com',
+            'first_name': 'org',
+            'last_name': 'admin'
+        }
+    },
+    '4': {
+        'account_number': 6089726,
+        'user': {
+            'username': 'notifications_admin',
+            'email': 'notifications_admin@redhat.com',
+            'first_name': 'notifications',
+            'last_name': 'admin'
+        }
+    }
+}
+
+# The entitlements server in ephemeral provides this data for each user
+ENTITLEMENTS = {
+    'insights': {'is_entitled': True, 'is_trial': False}
+}
+
+
+app = Flask(__name__)
+app.config['UPLOAD_FOLDER'] = FILE_CACHE_DIR
+
+
+if not os.path.exists(FILE_CACHE_DIR):
+    os.makedirs(FILE_CACHE_DIR)
```

Comment:
Suggestion:

passing `exist_ok` can replace the if guard.

```suggestion
os.makedirs(FILE_CACHE_DIR, exist_ok=True)
```